src/js/ble_uuid_util.js

```diff
@@ -36,8 +36,23 @@
 
 module.exports.removeDashes = function(uuid) {
   if (uuid) {
-    uuid = uuid.replace(/-/g, '');
+    uuid = uuid.split('-').join('');
   }
 
   return uuid;
 };
+
+module.exports.refineUuid = function(str, joint) {
+  str = String(str);
+  var len = str.length;
+  var arr = [];
+  var i = 0;
+
+  while (i < len - (len % 2)) {
+    arr.push(str.charAt(i) + str.charAt(i + 1));
+    i = i + 2;
+  }
+
+  if (i < len) arr.push(str.charAt(i));
+  return arr.reverse().join(joint);
```

Comment:
What about creating the string on-the-fly without using an array? Should be faster and use less memory. You can also iterate back to front.